core-rust-bridge/src/main/java/com/radixdlt/statecomputer/commit/LedgerHeader.java

```diff
@@ -0,0 +1,84 @@
+ * and any representation, warranty or undertaking in any form whatsoever, whether express
+ * or implied, to any entity or person, including any representation, warranty or
+ * undertaking, as to the functionality security use, value or other characteristics of
+ * any distributed ledger nor in respect the functioning or value of any tokens which may
+ * be created stored or transferred using the Work. The Licensor does not warrant that the
+ * Work or any use of the Work complies with any law or regulation in any territory where
+ * it may be implemented or used or that it will be appropriate for any specific purpose.
+ *
+ * Neither the licensor nor any current or former employees, officers, directors, partners,
+ * trustees, representatives, agents, advisors, contractors, or volunteers of the Licensor
+ * shall be liable for any direct or indirect, special, incidental, consequential or other
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.statecomputer.commit;
+
+import com.radixdlt.lang.Option;
+import com.radixdlt.sbor.codec.CodecMap;
+import com.radixdlt.sbor.codec.StructCodec;
+import com.radixdlt.utils.UInt64;
+
+public record LedgerHeader(
```

Comment:
similarly, why don't we reuse `com.radixdlt.consensus.LedgerHeader`?

Reply:
![image](https://user-images.githubusercontent.com/119413677/223723774-b1895464-7b00-4f92-9362-b8bbcfe30963.png)

Reply:
It is interesting though.

We are simply moving more and more responsibilities into rust (as planned, it seems). And suddenly rust starts caring about e.g. safety/consistency checks against ledger proof, so it can no longer be opaque to it. It kinda enters the "consensus" territory. But all the classes handling consensus are in their well-separated, java-only part of codebase.

Maybe this is not wrong - we do have some "duplicated" structures, yes, but their purpose is "DTO on java-rust bridge". A purpose quite different than "business logic for consensus purposes". Technically, these classes could evolve independently (e.g. the DTO could keep some of its parts as opaque `byte[]`).

Reply:
Hmh, it feels a bit weird IMO to have those DTOs for domain objects. I don't think they should ever evolve independently.

How about this: we mentally move the "ownership" of those domain objects to Rust and then have their Java counterparts living in `core-rust-bridge`, and use those objects in consensus etc?

Reply:
They kinda already evolve independently, in some details, e.g.:
- our domain objects are annotated with `@JsonProperty` and `@DsonOutput`
- our domain objects sometimes have extra "logic only" fields/caches (e.g. `BFTValidatorSet.totalPower`)
- our bridge DTOs use rust-friendly types (`Option` instead of `Optional`, `UInt64`instead of `long`, and more)

I know it is tempting to have "just one, true, ultimate representation of a `LedgerProof`, everywhere across all layers", but it is sometimes either a trap, or an unattainable ideal :(

Reply:
It's tempting indeed :) and I think all of your concerns could be addressed. Anyway, happy to go with a separate dto for now

Reply:
Ok, let me start a separate PR, in which I try to migrate the existing `LedgerHashes` class (an easy example of a "duplicated" domain vs dto class).

I will move the `com.radixdlt.consensus.LedgerHashes` into `com.radixdlt.statecomputer.commit` (i.e. replacing the existing `record LedgerHashes` from there), and update all the code that refers it. We will see how that looks (and decide whether to move on with heavier `LedgerHeader`, `LedgerProof` etc) 👍 

This PR is unblocked in this regard 🟢